tuning/data/data_processors.py

```diff
@@ -349,23 +349,10 @@ def process_dataset_configs(
         self, dataset_configs: List[DataSetConfig], **kwargs
     ) -> Union[Dataset, IterableDataset]:
         train_dataset = None
+        state = PartialState()
 
-        if torch.distributed.is_available() and torch.distributed.is_initialized():
-            if torch.distributed.get_rank() == 0:
-                logger.info("Processing data on rank 0...")
-                train_dataset = self._process_dataset_configs(dataset_configs, **kwargs)
-            else:
-                train_dataset = None
-
-            # Use broadcast_object_list to share the dataset object across ranks
-            # TODO: Check if torch.distributed.barrier() is called in broadcast_object_list()
-            # See https://github.com/pytorch/pytorch/issues/56142
-            # for why the list is shared like this
-            to_share = [train_dataset]
-            torch.distributed.broadcast_object_list(to_share, src=0)
-            train_dataset = to_share[0]
-        else:
-            logger.info("Processing data...")
+        # The main_process_first context ensures that the main process runs first
+        with state.main_process_first():
```

Comment:
```suggestion
        with state. local_main_process_first():
```

This will be relevant when each hf cache folder is different for each node, for instance, we point to `/dev/shm`.

Reply:
@willmj +1 can you test by using local main process first and we can ensure this works for both multi and single node.